packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -240,4 +251,109 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
       LIMIT 1;
     `);
   }
+
+  async getUsersWhoHaveMadeDialogues(): Promise<DbUser[]> {
+    return this.getRawDb().any(`
+      WITH all_dialogue_authors AS
+        (SELECT (UNNESTED->>'userId') AS _id
+            FROM "Posts" p, UNNEST("coauthorStatuses") unnested
+            WHERE p."collabEditorDialogue" IS TRUE 
+            AND p."draft" IS FALSE
+        UNION
+        SELECT p."userId" as _id
+            FROM "Posts" p
+            WHERE p."collabEditorDialogue" IS TRUE
+            AND p."draft" IS FALSE
+        )
+      SELECT u.*
+      FROM "Users" u
+      INNER JOIN all_dialogue_authors ON all_dialogue_authors._id = u._id
+    `)
+  }
+
+  async getUsersWhoHaveOptedInToDialogueFacilitation(): Promise<DbUser[]> {
+    return this.getRawDb().any(`
+        SELECT *
+        FROM "Users" u
+        WHERE u."optedInToDialogueFacilitation" IS TRUE
+    `)
+  }  
+
+  async getUsersTopUpvotedUsers(userId:string): Promise<UpvotedUser[]> {
+
+    const user = await getUser(userId)
+    const karma = user ? user.karma : 0
```

Comment:
A better way to express this is `user?.karma ?? 0`.  (Also, fun fact: `karma` is one of those fields which is incorrect nullable in the database, and as a result, we've had a few bugs over the years because numeric comparisons always return `false` when comparing a number to `null` or `undefined`.)

Reply:
Glad y'all fixed that!